vscode-lean4/src/leanclient.ts

```diff
@@ -489,6 +481,53 @@ export class LeanClient implements Disposable {
         return this.running ? this.client?.initializeResult : undefined
     }
 
+    private async checkToolchainVersion(folderUri: Uri) : Promise<Date | undefined> {
+        const lakefile = Uri.joinPath(folderUri, 'lakefile.lean').toString()
+        if (!fs.existsSync(new URL(lakefile))) {
+            return undefined;
+        }
+
+        // see if we have a well known toolchain label that corresponds
+        // to a known date like 'leanprover/lean4:nightly-2022-02-01'
+        const toolchainVersion = await readLeanVersion(folderUri);
```

Comment:
I don't think this version check is necessary anymore (since we got rid of the extremely expensive `lake serve` check).  But if you want to keep it, then at least use a regexp please to parse the toolchain version:
```javascript
const match = /^leanprover\/lean4:nightly-(\d+)-(\d+)-(\d+)$/.exec(toolchainVersion);
if (match) {
  return new Date(match[1], match[2] - 1, match[3]);
}
```